src/abstractions/serialization/KiotaSerializer.Serialization.cs

```diff
@@ -25,12 +27,25 @@ public static partial class KiotaSerializer
     /// <typeparam name="T">Type of the object to serialize</typeparam>
     /// <param name="contentType">Content type to serialize the object to </param>
     /// <param name="value">The object to serialize.</param>
+    /// <param name="serializeOnlyChangedValues">By default you'll only get the changed properties.</param>
     /// <returns>The serialized representation as a stream.</returns>
-    public static Stream SerializeAsStream<T>(string contentType, T value) where T : IParsable
+    public static Stream SerializeAsStream<T>(string contentType, T value, bool serializeOnlyChangedValues = true) where T : IParsable
     {
+        bool restoreInitializationCompleted = false;
+        if(!serializeOnlyChangedValues && value is IBackedModel backedModel)
```

Comment:
That's only going to work for the root object. 
I think that instead of doing that we should update the event listeners to switch the flag and call into the existing receiver instead.